packages/validator/src/services/attestation.ts

```diff
@@ -122,7 +166,10 @@ export class AttestationService {
 
     await Promise.all(
       duties.map(async ({duty}) => {
+        // TODO:  duty <> attestation consistency check required? slot and index are checked by lighthouse
+        // see https://github.com/sigp/lighthouse/blob/319cc61afeb1dbf3692e280dfa18e7b455542b16/validator_client/src/attestation_service.rs#L367
         const index = duty.committeeIndex;
+        // TODO: mutating is not required if attestation grouping is disabled as actual committee index is already set
```

Comment:
not a big deal here that we also mutate attestation if grouping is disabled as this would not be necessary since the committee index is already correct. What do you think @dapplion?

Reply:
Yeah but otherwise complicates the logic more to save an object allocation, which is whatever